scipy/stats/tests/test_sampling.py

```diff
@@ -5,50 +5,66 @@
 import numpy as np
 from numpy.testing import assert_allclose, assert_equal, suppress_warnings
 from numpy.lib import NumpyVersion
-from scipy.stats import TransformedDensityRejection, DiscreteAliasUrn
+from scipy.stats import (
+    TransformedDensityRejection,
+    DiscreteAliasUrn,
+    NumericalInversePolynomial
+)
 from scipy.stats import UNURANError
 from scipy import stats
+from scipy import special
 from scipy.stats import chisquare, cramervonmises
-from scipy.stats._distr_params import distdiscrete
+from scipy.stats._distr_params import distdiscrete, distcont
 
 
 # common test data: this data can be shared between all the tests.
 
 
 # Normal distribution shared between all the continuous methods
-class common_cont_dist:
-    # Private methods are used for performance. Public methods do a lot of
-    # validations and expensive numpy operations (like broadcasting), hence,
-    # slowing down some of the methods significantly. We can use private
-    # methods as we know that the data is guaranteed to be inside the domain
-    # and a scalar (so, no masking bad values and broadcasting required).
-    pdf = stats.norm._pdf
-    dpdf = lambda x: -x * stats.norm._pdf(x)  # noqa: E731
-    cdf = stats.norm._cdf
+class StandardNormal:
+    def pdf(self, x):
+        return np.exp(-0.5 * x*x)
+
+    def dpdf(self, x):
+        return -x * np.exp(-0.5 * x*x)
+
+    def cdf(self, x):
+        return special.ndtr(x)
 
 
 # A binomial distribution to share between all the discrete methods
-class common_discr_dist:
-    params = (10, 0.2)
-    pmf = partial(stats.binom._pmf, *params)
-    cdf = partial(stats.binom._cdf, *params)
+class Binomial:
+    def __init__(self, n, p):
+        self.n = n
+        self.p = p
+
+    def pmf(self, k):
+        return self.p**k * (1-self.p)**(self.n-k)
+
+    def cdf(self, k):
+        k = np.asarray(k)
+        return stats.binom._cdf(k, self.n, self.p)
+
+    def support(self):
+        return 0, self.n
 
 
 all_methods = [
-    ("TransformedDensityRejection", {"dist": common_cont_dist}),
-    ("DiscreteAliasUrn", {"dist": [0.02, 0.18, 0.8]})
+    ("TransformedDensityRejection", {"dist": StandardNormal()}),
+    ("DiscreteAliasUrn", {"dist": [0.02, 0.18, 0.8]}),
+    ("NumericalInversePolynomial", {"dist": StandardNormal()})
 ]
 
 
 bad_pdfs_common = [
     # Negative PDF
-    (lambda x: -x, UNURANError, r"50 : PDF\(x\) < 0.!"),
+    (lambda x: -x, UNURANError, r"..."),
```

Comment:
why did you change this?

Reply:
> why did you change this?

Different generators throw different errors:

TDR throws `50 : PDF(x) < 0.!` while PINV throws `51 : PDF(center) <= 0.!`

In general, we just want to make sure that an internal error occurs when invalid callbacks are passed to UNU.RAN functions. In that case, it's OK not to validate the message. If you agree, I will mention this point as an inline comment.

Reply:
Understood!